infra/terraform/file_transfer.go

```diff
@@ -0,0 +1,78 @@
+package terraform
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/gravitational/trace"
+)
+
+/*
+ FIXME: there are still few fundamental issues with this approach to SSH,
+ and it need be replaced with command by command execution
+
+ 1. ability to kill hung remote command (i.e. ./gravity wizard)
+    otherwise robotest completes but node is unsusable as there's still background process running
+
+ 2. waiting for boostrap script to complete (or maybe replacing its behind-the-scenes with more explicit preparation here)
+
+ 3. being sure which command exactly failed and avoiding monstrous && style pipelining
+
+ 4. support for SCP for local tarballs
+*/
+
+// TransferFile takes file URL which may be S3 or HTTP or local file and transfers it to the machine
+// fileUrl - file to download, could be S3:// or http(s)://
+// command - to run out of installer package
+func (t *terraform) makeRemoteCommand(fileUrl, command string) (string, error) {
+	u, err := url.Parse(fileUrl)
+	if err != nil {
+		return "", trace.Wrap(err, "Parsing %s", fileUrl)
+	}
+
+	var fetchCmd string
```

Comment:
I would declare this right before the switch where it's used. In general, I think it's a good idea to declare vars as close to their first usage location as possible.